Src/Web/Web.Shared.Net/SyntheticUserAgentTelemetryInitializer.cs

```diff
@@ -49,19 +51,26 @@ protected override void OnInitializeTelemetry(HttpContext platformContext, Reque
             {
                 if (platformContext != null)
                 {
-                    var request = platformContext.GetRequest();
-
-                    if (request != null && !string.IsNullOrEmpty(request.UserAgent))
+                    if (platformContext.Items.Contains(SyntheticSourceNameKey))
```

Comment:
I couldn't find the reason how checking requestTelemetry will improve perf even. Even for the requestTelemetry, initializer must be called to populate the field, and by doing that, the value is cached to platformConext.Items.